### PR TITLE
Add new login page.

### PR DESCRIPTION
**Pull request by mayankvishwakarma012**
**Linked Work Items**
- https://internship-splitwise.atlassian.net/jira/software/projects/IS/issues/IS-42

**Checklist**
- [x] Code adheres to **[Git and Pull Request Guidelines]**
- [x] Code adheres to **[Angular Guidelines]**
- [x] Code adheres to **[Front-end Checklist]**

### CODE REVIEW

File: src/app/user/login/login.component.html [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+
+
+            <div class="form-control">
+                <button type="submit">Login</button>
+            </div>
+
+            <div class="form-control"><a href="#">Forgot your password</a></div>
+        </form>
+    </div>
+</div>
```
Comment (ravipromact) [inline src/app/user/login/login.component.html:44]: Add comments

File: src/app/app.component.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+$green: #04a87f;
+.background-green {
+    color: $green;
+}
```
Comment (ravipromact) [inline src/app/app.component.scss]: Create one common scss file for colors variable where you can declare any color variable and re-use that file throughout the app

File: src/app/user/login/login.component.scss [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            .inner {
+                width: 80%;
+            }
+        }
+        @media screen and (max-width:600px) {
+            .inner {
+                width: 90%;
+            }
+        }
+    }
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss:68]: Use variables for colors

File: src/app/user/login/login.component.ts [hunk 4]
```diff
@@ -1,10 +1,25 @@
...
+  ngOnInit(): void {
+    this.loginForm = this.formBuilder.group({
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', Validators.required]
+    });
+  }
+
+  onSubmit() {
+    console.log(this.loginForm.value);
+  }
```
Comment (ravipromact) [inline src/app/user/login/login.component.ts]: In onSubmit() remove console.log, after removing console.log if the method is empty or it's not in use then remove the method also

File: src/app/user/login/login.component.scss [hunk 5]
```diff
@@ -0,0 +1,73 @@
+    // login form css
+    .container {
+        width: 100vw;
+        padding: 20px 0;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        // inner form div
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss]: Add space above the comments for better readability

File: src/app/user/login/login.component.scss [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        align-items: center;
+        justify-content: center;
+        // inner form div
+        .inner {
+            width: 40%;
+            padding: 20px;
+            display: block;
+            border: 1px solid #888888;
+            box-shadow: 2px 3px 2px #888888;
+            // form container design
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss]: Add space above the comments for better readability

File: src/app/user/login/login.component.scss [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            .form-container {
+                .form-control {
+                    border: none;
+                    display: flex;
+                    flex-direction: column;
+                    // form text box style
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss]: Add space above the comments for better readability

File: src/app/user/login/login.component.scss [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                        width: 100%;
+                        padding: 8px;
+                        margin: 8px 0;
+                        display: inline-block;
+                        border: 1px solid white;
+                        border-bottom: 1px solid #ccc;
+                        border-radius: 4px;
+                        box-sizing: border-box;
+                    }
+                    // form button style
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss]: Add space above the comments for better readability

File: src/app/user/login/login.component.scss [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+                        text-decoration: none;
+                        color: #18A582;
+                         :hover {
+                            color: #04a87f;
+                        }
+                    }
+                }
+            }
+        }
+        // responsive view for diffrent screen breakpoints
```
Comment (ravipromact) [inline src/app/user/login/login.component.scss]: Add space above the comments for better readability

File: src/app/app.component.html [hunk 10]
```diff
@@ -1,11 +1,49 @@
...
+</div>
+
+<!-- show component in routing -->
 <router-outlet></router-outlet>
+
+
+<!-- footer  -->
+<footer class="container py-3 my-4">
+    <hr>
+
```
Comment (ravipromact) [inline src/app/app.component.html]: Remove unwanted space

**Comment by ravipromact** [comment]
1. Title of pr is not as per guidelines, so 'Feature login' can be 'Add new login page'
2. All commits are not as per guidelines, for example 'Login page scss update' can be written as 'Update the styles for login page'

File: src/app/user/login/login.component.html (hunk 1 above)
Comment (mayankvishwakarma012) [reply src/app/user/login/login.component.html:44]: Done sir

File: src/app/app.component.scss (hunk 2 above)
Comment (mayankvishwakarma012) [reply src/app/app.component.scss]: Created a common scss folder and added color.scss file.

File: src/app/user/login/login.component.scss (hunk 3 above)
Comment (mayankvishwakarma012) [reply src/app/user/login/login.component.scss:68]: Replaced color names with color variables.

File: src/app/user/login/login.component.ts (hunk 4 above)
Comment (mayankvishwakarma012) [reply src/app/user/login/login.component.ts]: Removed the onSubmit method.

File: src/app/app.component.scss (hunk 2 above)
Comment (ravipromact) [reply src/app/app.component.scss]: You have not imported the variables file, i can still see the $green variable declaration

**Comment by ravipromact** [comment]
1. Don't commit every single change, complete all the changes and then commit
2. You've created a variable file in app component as common-css which is wrong, please create it under assets folder and use it

**Review by ravipromact — APPROVED** [review]

**Pull request merged**